families/identity/sawtooth_identity/src/handler.rs

```diff
@@ -0,0 +1,460 @@
+        let address = self.get_policy_address(name);
+        self.get_state_data(&address)
+
+    }
+
+    fn get_role_state(&mut self, name: &str) -> Result<Option<Vec<u8>>, ApplyError> {
+        let address = self.get_role_address(name);
+        self.get_state_data(&address)
+
+    }
+
+    fn get_state_data(&mut self, address: &str) -> Result<Option<Vec<u8>>, ApplyError> {
+            self.context.get_state(vec![address.to_string()]).map_err(|err| {
+                 #[cfg(not(target_arch = "wasm32"))]
+                warn!("Invalid transaction: Failed to load state: {:?}", err);
+                ApplyError::InvalidTransaction(format!("Failed to load state: {:?}", err))
+            })
+        }
+
+    fn set_policy(&mut self, policy_name: &str, policy_list: &PolicyList) -> Result<(), ApplyError> {
+        let address = self.get_policy_address(policy_name);
+        let data = protobuf::Message::write_to_bytes(policy_list).map_err(|err| {
+            ApplyError::InternalError(format!("Failed to serialize PolicyList: {:?}", err))
+        })?;
+
+        self.set_state(policy_name, &address, data)
+    }
+
+    fn set_role(&mut self, role_name: &str, role_list: &RoleList) -> Result<(), ApplyError> {
+        let address = self.get_role_address(role_name);
+        let data = protobuf::Message::write_to_bytes(role_list).map_err(|err| {
+            ApplyError::InternalError(format!("Failed to serialize RoleList: {:?}", err))
+        })?;
+        self.set_state(role_name, &address, data)
+
+    }
+
+    fn set_state(&mut self, name: &str, address: &String, data: Vec<u8>) -> Result<(), ApplyError> {
+        let mut state_entries = HashMap::new();
+        state_entries.insert(address.clone(), data);
+        self.context.set_state(state_entries).map_err(|err| {
+             #[cfg(not(target_arch = "wasm32"))]
+            warn!("Failed to set {} at {}", name, address);
+            ApplyError::InternalError(format!("Unable to save role {}", name))
+        })?;
+         #[cfg(not(target_arch = "wasm32"))]
+        debug!("Set: \n{:?}", name);
+
+         #[cfg(not(target_arch = "wasm32"))]
+        self.context.add_event("identity/update".to_string(),
+                         vec![("updated".to_string(), name.to_string())], &vec![])
+                         .map_err(|err| {
+                              #[cfg(not(target_arch = "wasm32"))]
+                             warn!("Failed to add event {}", name);
+                             ApplyError::InternalError(format!("Failed to add event {}", name))
+                         })?;
+        Ok(())
+    }
+
+
+    fn get_policy_address(&mut self, policy_name: &str) -> String {
+            format!("{}{}{}", IDENTITY_NAMESPACE.to_string(), IdentityState::POLICY_PREFIX.to_string(),
+                    self.to_hash(policy_name)[..62].to_string())
+        }
+
+    fn get_role_address(&mut self, role_name: &str) -> String {
```

Comment:
Likewise, possibly a clearer way to represent this: https://github.com/hyperledger/sawtooth-core/blob/master/validator/src/state/identity_view.rs#L182-L207